lightningd/dual_open_control.c

```diff
@@ -1252,21 +1251,29 @@ static void handle_peer_tx_sigs_sent(struct subd *dualopend,
 		return;
 	}
 
-	if (psbt_finalize(cast_const(struct wally_psbt *, psbt))) {
-		wtx = psbt_final_tx(NULL, psbt);
+	inflight = channel_current_inflight(channel);
+	if (psbt_finalize(cast_const(struct wally_psbt *,
+				     inflight->funding_psbt))
```

Comment:
psbt_finalize takes a non-const because it changes it.  That means inflight->funding_psbt needs to be non-const.

casting-away const is required sometimes because the function being called doesn't actually need const (e.g. strstr()).  But it's technically illegal to modify something via a const pointer, even if you cast it.

Reply:
Nice, we can just remove the `const_cast` here then, `inflight->funding_psbt`'s already 'non-const'.